### DISCUSSION TITLE
Admin ui screen for bulk role management

### DISCUSSION THREAD

**Discussion by bgiromini** [ServiceStack/Discuss#50] [Ideas]
Would be nice to have admin ui screens where I can manage roles. More specifically:

1. Add/Remove/Edit roles
2. Bulk add/delete users in a role. 

Would also like to have methods somewhere that could return all the users in a specific role like IAuthRepository.GetUsersInRoleAs<AppUser>("Admin")

**Comment by mythz** [comment]
There's no concept of "Adding Roles" in an Auth Repository, the list of available Roles in the User Admin comes from your Application's [Auth Attributes](https://docs.servicestack.net/authentication-and-authorization#authorization-attributes) in its C# source code, i.e. not from your Auth Repository datastore.

The ability to fetch all users with a role is be highly dependent on your Auth Repository, i.e. this can really only be efficiently implemented in Auth Repositories which maintain roles in a distinct table like `OrmLiteAuthRepository` with `UseDistinctRoleTables = true`, in which case you can query its tables directly, e.g: `Db.Select<UserAuthRole>(x => x.Role == role)`.

**Comment by bgiromini** [comment]
Thanks for the information. In my use case the roles are stored in userauthrole due to using MicrosoftAuthProvider so that simplifies things.
